models/pipeline.js

```diff
@@ -19,7 +19,7 @@ const CREATE_MODEL = {
         .required(),
 
     rootDir: Scm.rootDir,
-    autoKeysGeneration: Joi.boolean().optional()
+    autoKeysGeneration: Joi.boolean().optional().description('Flag to generate and add the public deploy key for a specific scm (used for private repos)')
```

Comment:
strictly not only for private repos. Any action which needs auth, eg: git clone for non-public SCM (Git hub enterprise vs github.com) needs deploy key.
Also is needed when build wants to commit back to SCM (for public repos)